website/docs/guides/orchestration/custom-cicd-pipelines/1-creating-custom-cicd-pipelines.md

```diff
@@ -0,0 +1,431 @@
+
+Jobs are executed on [runners](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#runners), which are virtual servers. The runners come pre-configured with Ubuntu Linux, macOS, or Windows. That means the commands you execute are determined by the operating system of your runner. You’ll see how this comes into play later in the setup, but for now just remember that your code is executed on virtual servers that are, typically, hosted by the code hosting platform.
+
+![Diagram of how pipelines work](/img/guides/orchestration/custom-cicd-pipelines/pipeline-diagram.png)
+
+Please note, runners hosted by your code hosting platform provide a certain amount of free time. After that, billing charges may apply depending on how your account is setup. You also have the ability to host your own runners. That is beyond the scope of this article, but checkout the links below for more information if you’re interested in setting that up:
+
+- Repo-hosted runner billing information:
+  - [GitHub](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions)
+  - [GitLab](https://docs.gitlab.com/ee/ci/pipelines/cicd_minutes.html)
+- Self-hosted runner information:
+  - [GitHub](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners)
+  - [GitLab](https://docs.gitlab.com/runner/)
+
+Additionally, if you’re using the free tier of GitLab you can still follow this guide, but it may ask you to provide a credit card to verify your account. You’ll see something like this the first time you try to run a pipeline:
+
+![Warning from GitLab showing payment information is required](/img/guides/orchestration/custom-cicd-pipelines/gitlab-cicd-payment-warning.png)
+
+
+
+## How to setup pipelines
+
+This guide provides details for multiple code hosting platforms. Where steps are unique, they are presented without a selection option. If code is specific to a platform (i.e. GitHub, GitLab) you will see a selection option for each.
+
+Pipelines can be triggered by various events. The [dbt Cloud webhook](https://docs.getdbt.com/docs/dbt-cloud/using-dbt-cloud/cloud-enabling-continuous-integration) process already triggers a run if you want to run your jobs on a merge request, so this guide focuses on running pipelines for every push and when PRs are merged. Since pushes happen frequently in a project, we’ll keep this job super simple and fast by linting with SQLFluff. The pipeline that runs on merge requests will run less frequently, and can be used to call the dbt Cloud API to trigger a specific job. This can be helpful if you have specific requirements that need to happen when code is updated in production, like running a `--full-refresh` on all impacted incremental models. 
+
+Here’s a quick look at what this pipeline will accomplish:
```

Comment:
I'm wondering if it would be better to separate each pipeline entirely? Like have pipeline 1 be entirely about SQLFluff and pipeline 2 be entirely about kicking off a cloud job on merge

Reply:
they are fully split in the guide, however in GitLab you define all your actions in a single file. That's why you still see SQLFluff in there for the merge. It's more clear in GitHub because they exist in difference files

Reply:
actually, I think I'll have to change more than just the code to make this change. This assumes people already have the structure setup. I'll re-work it a bit